tests/serverpod_test_server/test_e2e/record_types_test.dart

```diff
@@ -529,31 +545,115 @@ void main() {
       );
       expect(result.namedSubRecord.$2, 4.5);
     });
+
+    test(
+        'Given the test server, when a `null` complex nested positioned record is sent to the server, then it is returned verbatim',
+        () async {
+      var record = (
+        (1, '2'),
+        namedSubRecord: (
+          SimpleData(num: 3),
+          4.5,
+        )
+      );
+
+      var result = await client.recordParameters
+          .returnListOfNestedPositionalAndNamedRecord([record, record]);
+
+      expect(result, hasLength(2));
+      expect(
+        result.first.$1,
+        (1, '2'),
+      );
+      expect(
+        result.first.namedSubRecord.$1,
+        isA<SimpleData>().having((d) => d.num, 'num', 3),
```

Comment:
All these assertions are a bit complicated (using `isA`), because models do not implement equality.

So the only way to test combinations with data classes is to unwrap them manually (instead of being able to assert something like `expect(response, emitsInOrder(requestParameters))`…)